src/diffpy/utils/diffraction_objects.py

```diff
@@ -36,26 +36,68 @@ def _setter_wmsg(attribute):
 
 class DiffractionObject:
     def __init__(
-        self, name=None, wavelength=None, scat_quantity=None, metadata=None, xarray=None, yarray=None, xtype=None
+        self,
+        xarray,
+        yarray,
+        xtype,
+        wavelength,
+        scat_quantity="",
+        name="",
+        metadata={},
     ):
-        if name is None:
-            name = ""
-        self.name = name
-        if metadata is None:
-            metadata = {}
-        self.metadata = metadata
-        if xtype is None:
-            xtype = ""
+
+        self._id = uuid.uuid4()
+        self.input_data(xarray, yarray, xtype, wavelength, scat_quantity, name, metadata)
+
+    def input_data(self, xarray, yarray, xtype, wavelength, scat_quantity="", name="", metadata={}):
+        """
+        Insert a new scattering quantity into the scattering object.
```

Comment:
Improved docstrings, previous version:

```
        f"""
        insert a new scattering quantity into the scattering object
        Parameters
        ----------
        xarray array-like of floats
          the independent variable array
        yarray array-like of floats
          the dependent variable array
        xtype string
          the type of quantity for the independent variable from {*XQUANTITIES, }
        metadata, scat_quantity, name and wavelength are optional.  They have the same
        meaning as in the constructor. Values will only be overwritten if non-empty values are passed.
        Returns
        -------
        Nothing.  Updates the object in place.
        """
```

Reply:
If we make it private we may want to move the docstring to the constructor?  Also, then change "insert a new scattering object blah blah, to instantiate a new scattering object or sthg.

Reply:
done